tools/deployer_crd_upgrade.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+#description  :This script converts existing Search Head Cluster CRs to deployer CR
+#              and search head cluster CR due to the introduction of the deployer CRD
+#
+# Usage:   ./deployer_crd_upgrade.sh upgrade <shc_yaml_file> <namespace_of_the_SHC_CR> <target_folder_for_yamls>
+# We do the following actions to facilitate the upgrade:
+# 1. Generate the modified yamls for the SHC and deployer CRs in the target folder
+# 2. Deploy the new deployer CR
+# 3. Deploy the new SHC, if modified
+#
+# Requirements
+# This script uses JQ
+
+#############################################
+# Helper Functions to setup the environment
+#############################################
+
+trap '{ echo "Script interrupted by admin" ; exit 1; }' INT
+
+err() {
+	echo -e "\n$(date +'%Y-%m-%d-%H:%M:%S%z') - ERROR - $*"
+	echo -e "\n********************** Update canceled **********************"
+	exit 1
+}
+
+prereqs() {
+	jq --version >/dev/null 2>&1
+}
+
+usage() {
+	echo -e "\nUsage: $0 <full_path_of_shc_yaml> <namespace> <full_path_to_target_folder_for_new_yamls>"
+	echo -e "Facilitates the entire update including the following actions"
+    echo -e "# 1. Generate the modified yamls for the SHC and deployer CRs in the target folder"
+    echo -e "# 2. Deploy the new deployer CR"
+    echo -e "# 3. Deploy the new SHC, if modified"
+    echo -e "Please make sure to pass all the required arguments"
```

Comment:
nit: You may want to foot a header and footer, so that, when usage() clubbed with other errors, it will be easy to follow the error.

Reply:
This is an initial version of the script. Will be taking up the migration script next sprint.